.github/shared/test/doc-preview.test.js

```diff
@@ -0,0 +1,147 @@
+import { test, describe, expect } from "vitest";
```

Comment:
```suggestion
// @ts-check

import { test, describe, expect } from "vitest";
```

We try to enable ts-check even for tests, but you can disable if it gets too cumbersome.